vendor/k8s.io/kubernetes/test/e2e/kubectl/kubectl.go

```diff
@@ -511,7 +511,7 @@ var _ = SIGDescribe("Kubectl client", func() {
 
 			ginkgo.By("executing a command with run and attach with stdin")
 			// We wait for a non-empty line so we know kubectl has attached
-			runOutput := framework.NewKubectlCommand(ns, nsFlag, "run", "run-test", "--image="+busyboxImage, "--restart=OnFailure", "--attach=true", "--stdin", "--", "sh", "-c", "while [ -z \"$s\" ]; do read s; sleep 1; done; echo read:$s && cat && echo 'stdin closed'").
+			runOutput := framework.NewKubectlCommand(ns, nsFlag, "run", "run-test", "--image="+busyboxImage, "--restart=Never", "--attach=true", "--stdin", "--", "sh", "-c", "while [ -z \"$s\" ]; do read s; sleep 1; done; echo read:$s && cat && echo 'stdin closed'").
```

Comment:
this is not in upstream because oc is incompatible?

Reply:
Correct. As per the commit message for this change, `kubectl run` always results in a pod but `oc run` will create a deployment, job or pod depending on the value provided to `--restart`.

Reply:
cc @soltysh. I don't like incompatible oc/kubectl

Reply:
This should not be necessary now that we are at 1.18 levels in oc (https://github.com/openshift/oc/pull/351). Iow. this means `oc run` is equivalent to `kubectl run` and creates ONLY pods. 

> cc @soltysh. I don't like incompatible oc/kubectl

Tell me about it :sweat: I'm doing my best not to have those too many.

Reply:
@soltysh I can confirm that the latest version of oc is consistent with upstream and always creates a pod. I've dropped this carry. 

While verifying this, I noticed the help text for the `--restart` argument was outdated and posted a PR to fix it: https://github.com/kubernetes/kubernetes/pull/89803